modules/generic/meta-maintainers.nix

```diff
@@ -0,0 +1,61 @@
+# Test:
+#   ./meta-maintainers/test.nix
+{ lib, ... }:
+let
+  inherit (lib)
+    mkOption
+    mkOptionType
+    types
+    ;
+
+  maintainer = mkOptionType {
+    name = "maintainer";
+    check = email: lib.elem email (lib.attrValues lib.maintainers);
```

Comment:
This is already hinted at in your PR description, but having the `check` function tightly coupled to `lib.maintainers` means this option-type isn't useful to projects that have their own `maintainers` set.

Perhaps this should be designed similarly to `mkPackageOption`, such that it takes a `maintainers` set as a function argument?

Reply:
Good idea. Are you thinking of a `meta.isMaintainer` (`functionTo bool`) option?